### PR TITLE
update vue binding to use typescript and added vue types

### PR DESCRIPTION
**Pull request by ksyeo1010**
- vue js to ts
- added types to vue
- added support for vue 2 and vue 3
- changed from renderless component to mixin

### CODE REVIEW

File: demo/vue/src/components/VoiceWidget.vue [hunk 1]
```diff
@@ -69,36 +62,44 @@ export default {
...
+    initEngine: function (event: any) {
       this.factoryArgs.accessKey = event.target.value;
       this.isError = false;
       this.isLoaded = false;
       this.isListening = false;
-      this.$refs.porcupine.initEngine();
-    },
+      this.$porcupine.init(
+        this.factoryArgs,
+        PorcupineWorkerFactoryEn,
```
Comment (mrrostam) [inline demo/vue/src/components/VoiceWidget.vue:81]: why not

```suggestion
        this.factory,
```

File: binding/vue/src/porcupine_types.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ 
+/**
+* The language-specific worker factory, imported as { PorcupineWorkerFactory } from the 
+* @picovoice/porcupine-web-xx-worker series of packages, where xx is the two-letter language code.
+*/
+export interface PorcupineWorkerFactory extends Object {
+  create: (
+    accessKey: string, 
+    keywords: Array<PorcupineKeyword | string> | PorcupineKeyword | string,
+    keywordDetectionCallback?: CallableFunction,
```
Comment (mrrostam) [inline binding/vue/src/porcupine_types.ts:42]: it has nothing to do with this PR, but i noticed we have this `CallableFunction`s everywhere and I think it's better to make them more specific at some point, something like `(keywordLabel: string) => void`

Comment (ksyeo1010) [reply binding/vue/src/porcupine_types.ts:42]: Sure we should do this later on.

File: binding/vue/README.md [hunk 3]
```diff
@@ -12,7 +12,9 @@ applications.
```
Comment (mrrostam) [inline binding/vue/README.md:12]: let's update the main readme

File: binding/vue/README.md [hunk 4]
```diff
@@ -12,7 +12,9 @@ applications.
 
 ## Compatibility
```
Comment (mrrostam) [inline binding/vue/README.md:13]: # porcupine-web-vue

Renderless Vue component for Porcupine for Web. -> is this still a correct description for this binding?

File: binding/vue/README.md [hunk 5]
```diff
@@ -12,7 +12,9 @@ applications.
...
 
-This library is compatible with Vue 3.
+This library is compatible with Vue:
+- Vue.js 2.6.11+.
+- Vue.js 3.0.0+.
```
Comment (laves) [inline binding/vue/README.md:17]:
```suggestion
- Vue.js 2.6.11+
- Vue.js 3.0.0+
```

**Pull request merged**